__tests__/integration/helpers/wallet.helper.js

```diff
@@ -0,0 +1,236 @@
+    // Event listener
+    const handleNewTx = newTx => {
+      // Ignore this event if we didn't receive the transaction we expected.
+      if (newTx.tx_id !== txId) {
+        return;
+      }
+
+      // This is the correct transaction: resolving the promise.
+      resolveWithSuccess(newTx);
+    };
+    hWallet.on('new-tx', handleNewTx);
+
+    // Timeout handler
+    const timeoutPeriod = timeout || TX_TIMEOUT_DEFAULT;
+    setTimeout(async () => {
+      hWallet.removeListener('new-tx', handleNewTx);
+
+      // No need to respond if the event listener worked.
+      if (alreadyResponded) {
+        return;
+      }
+
+      /*
+       * If the timeout period passed and did not receive the new-tx event, probably the event
+       * was triggered even before this `waitForTxReceived` method was called.
+       * We'll try a last time to get the transaction data before rejecting this promise.
+       */
+      loggers.test.log(`Trying to solve timeout of ${timeoutPeriod}ms for ${txId}`);
+      const existingTx = hWallet.getTx(txId);
+      if (existingTx) {
+        return resolveWithSuccess(existingTx);
+      }
+
+      // Event listener did not receive the tx and it is not on local cache.
+      alreadyResponded = true;
+      reject(new Error(`Timeout of ${timeoutPeriod}ms without receiving tx ${txId}`));
+    }, timeoutPeriod);
+
+    async function resolveWithSuccess(newTx) {
+      const timeDiff = Date.now().valueOf() - startTime;
+      if (DEBUG_LOGGING) {
+        loggers.test.log(`Wait for ${txId} took ${timeDiff}ms.`);
+      }
+
+      if (alreadyResponded) {
+        return;
+      }
+      alreadyResponded = true;
+
+      /*
+       * Sometimes the tests happen too fast and it triggers timestamp protections on the fullnode.
+       * We will ensure there is some delay between transactions here.
+       */
+      await delay(50);
+
+      /*
+       * Confirming that the transaction is indeed on the wallet's history object before preoceeding
+       */
+      let txObj = hWallet.getTx(txId);
+      while (!txObj) {
+        if (DEBUG_LOGGING) {
+          loggers.test.warn(`Tx was not available on history. Waiting for 50ms and retrying.`);
+        }
+        await delay(50);
+        txObj = hWallet.getTx(txId);
+      }
+      resolve(newTx);
+    }
+  });
+}
+
+/**
+ * This method helps a tester to ensure the current timestamp of the next transaction will be at
+ * least one unit greater than the specified transaction.
+ *
+ * Hathor's timestamp has a granularity of seconds, and it does not allow one transaction to have a
+ * parent with a timestamp equal to its own.
+ *
+ * It does not return any content, only delivers the code processing back to the caller at the
+ * desired time.
+ *
+ * @param {HathorWallet} hWallet
+ * @param {string} txId
+ * @returns {void}
+ */
+export async function waitUntilNextTimestamp(hWallet, txId) {
+  const { timestamp } = hWallet.getTx(txId);
```

Comment:
All places that call this method already have the tx hash for other uses, but not all of them have the complete transaction object. For example:
```js
const { hash: delegateMintTxId } = await hWallet1.delegateAuthority( ... )
```

I felt it was a small price to re-calculate this in order to keep the test code as readable as possible, especially since other helper methods also require the `HathorWallet + tx hash` parameters.